tracer/test/Datadog.Trace.TestHelpers/TestHelper.cs

```diff
@@ -215,8 +215,20 @@ public ProcessResult RunSampleAndWaitForExit(MockTracerAgent agent, string argum
 
             using var helper = new ProcessHelper(process);
 
-            process.WaitForExit();
-            helper.Drain();
+            // this is _way_ too long, but we want to be v. safe
+            // the goal is just to make sure we kill the test before
+            // the whole CI run times out
+            var timeoutMs = (int)TimeSpan.FromMinutes(10).TotalMilliseconds;
+            var ranToCompletion = process.WaitForExit(timeoutMs) && helper.Drain(timeoutMs / 2);
+
+            if (!ranToCompletion && !process.HasExited)
+            {
+                var tookMemoryDump = TakeMemoryDump(process);
+                process.Kill();
+                // should we throw a skip exception on Linux as we don't have a memory dump?
+                throw new Exception($"The sample did not exit in {timeoutMs}ms. Memory dump taken: {tookMemoryDump}. Killing process.");
```

Comment:
Total nitpicking, but `600000ms` is a big number that you need to do math on to comprehend when reading the error. Could we do something smaller like `600s` or `10 mins` for readability?

level of care: low

Reply:
YEs, we could, though the number is almost irrelevent, so I'm inclined to remove it entirely 😉 I will slide into a separate PR, just to avoid another ci loop for now